### PR TITLE
DCP-4034: fix order based on portal api query

### PR DESCRIPTION
**Pull request by lilim-ebi**

### CODE REVIEW

File: python3/utils.py [hunk 1]
```diff
@@ -590,15 +590,15 @@ def split_filelist(filelist_string):
 
 def parse_file_search_result_line(line, accession, output_format):
     cols = line.split('\t')
-    data_acc = cols[0].strip()
-    sub_filelist = split_filelist(cols[1])
-    sub_md5list = split_filelist(cols[2])
+    data_acc = cols[-1].strip()
```
Comment (suranjayathilaka) [inline python3/utils.py:593]: what does the response look like? can u add an example here as a comment

Comment (lilim-ebi) [reply python3/utils.py:593]: This is the request to the API:
https://www.ebi.ac.uk/ena/portal/api/search?query=sample_accession=%22SAMEA1116773%22&result=read_run&fields=submitted_ftp,submitted_md5,sra_ftp,sra_md5,fastq_ftp,fastq_md5&limit=0

Response:
| submitted_ftp                                                                                                                                                               | submitted_md5                                | sra_ftp                                                       | sra_md5                               | fastq_ftp                                                                                                                | fastq_md5                              | run_accession |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------|---------------------------------------------------------------|---------------------------------------|--------------------------------------------------------------------------------------------------------------------------|----------------------------------------|---------------|
| ftp.sra.ebi.ac.uk/vol1/run/ERR251/ERR2512031/20104421_S5_L999_R1_001.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR251/ERR2512031/20104421_S5_L999_R2_001.fastq.gz | 5267a0aa15395983b08318af330bfe47;e351cea6ed9d2f45f2d5fc01238789e5 | ftp.sra.ebi.ac.uk/vol1/err/ERR251/001/ERR2512031             | 1cf4167dfebec580f3a9f8927c546cc7 | ftp.sra.ebi.ac.uk/vol1/fastq/ERR251/001/ERR2512031/ERR2512031_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/ERR251/001/ERR2512031/ERR2512031_2.fastq.gz | 950123b5264f6483040901575a8e8383;8bb209553d1c0292593524813cffb67f | ERR2512031   |

Comment (lilim-ebi) [reply python3/utils.py:593]: submitted_ftp	submitted_md5	sra_ftp	sra_md5	fastq_ftp	fastq_md5	run_accession
ftp.sra.ebi.ac.uk/vol1/run/ERR251/ERR2512031/20104421_S5_L999_R1_001.fastq.gz;ftp.sra.ebi.ac.uk/vol1/run/ERR251/ERR2512031/20104421_S5_L999_R2_001.fastq.gz	5267a0aa15395983b08318af330bfe47;e351cea6ed9d2f45f2d5fc01238789e5	ftp.sra.ebi.ac.uk/vol1/err/ERR251/001/ERR2512031	1cf4167dfebec580f3a9f8927c546cc7	ftp.sra.ebi.ac.uk/vol1/fastq/ERR251/001/ERR2512031/ERR2512031_1.fastq.gz;ftp.sra.ebi.ac.uk/vol1/fastq/ERR251/001/ERR2512031/ERR2512031_2.fastq.gz	950123b5264f6483040901575a8e8383;8bb209553d1c0292593524813cffb67f	ERR2512031

Comment (lilim-ebi) [reply python3/utils.py:593]: What happened is that the accession came first in the response until now, but it changed to being the last.

Comment (suranjayathilaka) [reply python3/utils.py:593]: right I meant add an this in the code file as a comment so the column numbers for the values make sense

File: python3/utils.py [hunk 2]
```diff
@@ -575,11 +573,16 @@ def get_result(accession):
     else:  # is_analysis(accession)
         return ANALYSIS_RESULT
 
+def get_result_accession(accession):
+    if is_run(accession) or is_experiment(accession) or is_sample(accession):
+        return 'run_accession'
+    else:  # is_analysis(accession)
```
Comment (suranjayathilaka) [inline python3/utils.py:579]: i think best to check for is_analysis() too

Comment (lilim-ebi) [reply python3/utils.py]: I added extra checks in these functions, but there are already checks in place at the start of the script so it shouldn't end up in any other branch.

File: python3/utils.py [hunk 3]
```diff
@@ -572,14 +570,28 @@ def get_file_fields(accession, aspera):
 def get_result(accession):
     if is_run(accession) or is_experiment(accession) or is_sample(accession):
         return RUN_RESULT
-    else:  # is_analysis(accession)
+    elif is_analysis(accession):
         return ANALYSIS_RESULT
+    else:
+        raise TypeError('Only runs, experiments, samples and analyses are allowed')
```
[the remark this answers is not in the document]
Comment (lilim-ebi) [reply python3/utils.py:576]: no, study can only be run from groupget and then read/analysis is downloaded

**Pull request merged**